grafana/provider.go

```diff
@@ -41,15 +45,19 @@ func Provider() terraform.ResourceProvider {
 }
 
 func providerConfigure(d *schema.ResourceData) (interface{}, error) {
+	splitAuth := strings.Split(d.Get("auth").(string), ":")
+	cli := cleanhttp.DefaultClient()
+	cli.Transport = logging.NewTransport("Grafana", cli.Transport)
 	client, err := gapi.New(
-		d.Get("auth").(string),
 		d.Get("url").(string),
+		gapi.Config{
+			BasicAuth: url.UserPassword(splitAuth[0], splitAuth[1]),
+			Client: cli,
+		},
 	)
```

Comment:
This doesn't account for the possibility that `auth` is an API key. What do you think about something like this instead...

```golang
	auth := strings.SplitN(d.Get("auth").(string), ":", 2)
	cli := cleanhttp.DefaultClient()
	cli.Transport = logging.NewTransport("Grafana", cli.Transport)
	cfg := gapi.Config{
		Client: cli,
	}
	if len(auth) == 2 {
		cfg.BasicAuth = url.UserPassword(auth[0], auth[1])
	} else {
		cfg.APIKey = auth[0]
	}
	client, err := gapi.New(d.Get("url").(string), cfg)
```